### DISCUSSION TITLE
Put request on snipe-it 422 error **Python Only**

### DISCUSSION THREAD

**Discussion by hgardnercsg-llcusa** [snipe/snipe-it#12735] [Q&A] [answered]
The put request on the sample api page. https://snipe-it.readme.io/reference/hardware-update gives a 422 error when attempting the hardware update by id. I tried on my personal page and the same error occured. It was 
{"message":"The given data was invalid.","errors":{"image":["The image must be a file of type: png, gif, jpg, jpeg, svg, bmp, svg+xml, webp."]}}"
I assume it has to do with specifically not allowing a null value on the image put request, but I'm just guessing. If anyone could assist with this issue that would be fantastic!!!

**Comment by snipe** [comment, marked as answer]
Hi - I've tried to reproduce this on the demo, and I kinda can't? Are you passing a value for the image field? Normally, you can just leave it out and it works fine. Can you give us screenshots, a curl request, etc so we can try to reproduce?

**Reply by snipe** [reply]
(Obviously obfuscate your bearer key, etc)

**Reply by hgardnercsg-llcusa** [reply]
I do apologize I have left out that it is in python only.
![image](https://user-images.githubusercontent.com/129088106/228004454-b633923d-b8cc-439a-9001-07c047538af0.png)

**Reply by hgardnercsg-llcusa** [reply]
Which is the same error being thrown to me in my script a similar PUT command with just fields changed

**Reply by snipe** [reply]
If you leave out the `image` field tho?

**Reply by hgardnercsg-llcusa** [reply]
okay okay, I apologize first off for jumping the gun. Here is my whole issue 
````````````````
import json
import requests

baseurl = 'http://sitename-snipe-dev/api/v1/hardware/'
headers = {
        "accept": "application/json",
        'Content-Type': 'application/json',
        "Authorization": "Bearer Redacted"
        }


response = requests.get(baseurl+'?sort=created_at&order=desc&model_id=13', headers=headers)

cost = 200

for i in json.loads(response.text)['rows']: ###iterate through all assets with old value
    i['purchase_cost'] = 200 #change old value
    #i.pop('image') ###tested removal of field
    #print (json.loads(response.text)['rows'][0]['image'])###prints None
    response2 = requests.put(baseurl+f"{i['id']}",json=i,headers=headers) ### Overwrites old entry BUT errors out with image still
    print(response2.text) #confirm the entry is overwritten
    ``````````````````````
    but I am still getting this PUT error. I have attempted on powershell and it works. If I need to I can write in powershell but if you see any error in my code or think this is something I'm doing wrong I would love assistance. I am still getting this error when even leaving out the image value/setting it to None.

**Reply by hgardnercsg-llcusa** [reply]
I just used the patch command to accomplish the task. I appreciate your assistance!!

**Reply by snipe** [reply]
Great, thanks for the follow up!
